deploy-web/src/components/layout/TransactionModal.tsx

```diff
@@ -3,18 +3,19 @@ import { Popup } from "../shared/Popup";
 import { Box, CircularProgress, Typography } from "@mui/material";
 
 type Props = {
+  state: "waitingForApproval" | "broadcasting";
   open: boolean;
-  onClose: () => void;
+  onClose?: () => void;
   children?: ReactNode;
 };
 
-export const TransactionModal: React.FunctionComponent<Props> = ({ open, onClose }) => {
+export const TransactionModal: React.FunctionComponent<Props> = ({ state, open, onClose }) => {
   return (
     <Popup
       fullWidth
       open={open}
       variant="custom"
-      title={<>Transaction Pending</>}
+      title={state === "waitingForApproval" ? <>Waiting for tx approval</> : <>Transaction Pending</>}
       actions={[]}
       onClose={onClose}
```

Comment:
Is the X on the popup still visible?

Reply:
Nope it's hidden when `onClose` is `undefined`